driver/src/main/java/org/neo4j/driver/v1/Path.java

```diff
@@ -61,7 +61,7 @@
     Node end();
 
     /** @return the number of segments in this path, which will be the same as the number of relationships */
-    long length();
+    int length();
```

Comment:
It gets unnatural from a Java POV (Java arrays are int based for example). By choosing longs we force the mental overhead of taking care there's no overflow etc to users instead of us taking care of that.  We talked this over and think if we need/want long indexing we'd rather duplicate the methods and have additional `longLength()` and `value( long )` methods.  These two should be enough. wdyt?